bin/live/pycbc_live_single_trigger_fits

```diff
@@ -0,0 +1,377 @@
+if args.output_fit_plot:
+    logging.info("Getting template counts from bank file")
+    bank_f = h5py.File(args.bank_file, 'r')
+    bank_array = DictArray(data={d : np.array(bank_f[d][:])
+                                 for d in bank_f})
+    for cparam, cineq, cthresh in zip(cut_params, cut_inequalities, cut_thresholds):
+        #Apply cuts to templates in this file
+        if cparam not in bank_f: continue
+        cineq_fn = inequality_string_to_function(cineq)
+        idx_keep = np.nonzero(cineq_fn(bank_array.data[cparam], cthresh))[0]
+        bank_array = bank_array.select(idx_keep)
+    logging.info("{} templates meet cut criteria".format(len(bank_array)))
+    logging.info("Plotting fits")
+    f_trigs = h5py.File(oput_file, "r")
+    bank_bin = np.array([tbins[d] for d in bank_array.data['template_duration']])
+    for ifo in args.ifos:
+        max_rate = 0
+        if not len(events[ifo].data[args.fit_param]): continue
+        maxstat = max(events[ifo].data[args.fit_param])
+        max_ct = 0
+        plotrange =  np.linspace(args.fit_threshold, 1.05 * maxstat)
+        event_bin = np.array([tbins[d] for d in events[ifo].data['template_duration']])
+        for bin_num, l, u, c, a in zip(range(args.num_duration_bins), tbins.lower(),
+                                       tbins.upper(), counts[ifo], alphas[ifo]):
+            inbin = event_bin == bin_num
+            parvals = events[ifo].data[args.fit_param][inbin]
+            if not len(parvals): continue
+            count_templates = np.count_nonzero(bank_bin == bin_num)
+            max_ct = max(max_ct, count_templates)
+            histcounts, edges = np.histogram(parvals, bins=50)
+            cum_counts = histcounts[::-1].cumsum()[::-1]
+            binlabel = r"%.3g - %.3g" % (l, u)
+            # histogram of fitted values
+            cum_rate = cum_counts / live_time[ifo] / count_templates
+            max_rate = max(max_rate, max(cum_rate))
+            cum_fit = c * trstats.cum_fit(args.fit_function, plotrange,
+                                          a, args.fit_threshold) / live_time[ifo] / count_templates
+            plt.semilogy(edges[:-1], cum_rate, linewidth=2,
+                         color=histcolors[bin_num], label=binlabel, alpha=0.6)
+            plt.semilogy(plotrange, cum_fit, "--", color=histcolors[bin_num],
+                         label=r"$\alpha = $%.2f" % a)
+        oput_plot = os.path.join(args.output_directory,
+                                 ifo + '-' + args.analysis_date + "-TRIGGER-FITS.png")
+        plt.grid()
+        plt.xlim(args.fit_threshold, 1.05 * maxstat)
+        plt.ylim(0.5 / live_time[ifo] / max_ct, 1.5 * max_rate)
+        plt.xlabel(args.fit_param)
+        plt.ylabel("Number of louder triggers per live time per applicable template")
+        plt.legend()
+        logging.getLogger().setLevel(logging.WARNING)
+        plt.savefig(oput_plot)
+        logging.getLogger().setLevel(logging.INFO)
```

Comment:
Why are you changing the log level in the middle of the script?

Reply:
This was in order to stop the *incredibly* verbose logging output that savefig does given a logging.info level. I think Tito wrote something to get around this issue, so will look for it and see if I can implement that here

Reply:
So you can declare your own logger at module level and then configure the log level _just_ for logs that originate from it.

Reply:
Thanks! - I found that collin had written a PR for this exact reason, and we have a shortcut to do exactly that

Reply:
In the end I removed this plotting functionality anyway as it was using a custom template bank which makes it unsuitable for wider use